src/stdlib_linalg_determinant.fypp

```diff
@@ -0,0 +1,302 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+module stdlib_linalg_determinant
```

Comment:
Could it be a submdodule of `stdblib_linalg`, such that users can use it as:
```fortran
use stdlib_linalg, only: det
```
instead of:
```fortran
use stdlib_linalg_determinant, only: det
```

This will avoid an enormous amount of `stdlib_linalg` modules.
Inconvenient: this approach will require the compilation of the a large module.

What do you think? What would be the best strategy for the users?

Reply:
This seems like a good approach, everything would fit together smoothly. This might require indeed a large header module but for the end user and even a developer who would like to pick just one method and work on it, it would be easier... I think.